controller/golden_files_test.go

```diff
@@ -66,48 +93,177 @@ func testableCompareWithGolden(update bool, goldenFile string, actualObj interfa
 	return nil
 }
 
+// findUUIDs returns an array of uniq UUIDs that have been found in the given
+// string
+func findUUIDs(str string) ([]uuid.UUID, error) {
+	pattern := "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}"
+	uuidRegexp, err := regexp.Compile(pattern)
+	if err != nil {
+		return nil, errs.Wrapf(err, "failed to compile UUID regex pattern %s", pattern)
+	}
+	uniqIDs := map[uuid.UUID]struct{}{}
+	var res []uuid.UUID
+	for _, uuidStr := range uuidRegexp.FindAllString(str, -1) {
+		ID, err := uuid.FromString(uuidStr)
+		if err != nil {
+			return nil, errs.Wrapf(err, "failed to parse UUID %s", uuidStr)
+		}
+		_, alreadyInMap := uniqIDs[ID]
+		if !alreadyInMap {
+			uniqIDs[ID] = struct{}{}
+			// append to array
+			res = append(res, ID)
+		}
+	}
+	return res, nil
+}
+
+// replaceUUIDs finds all UUIDs in the given string and replaces them with
+// "00000000-0000-0000-0000-000000000001,
+// "00000000-0000-0000-0000-000000000002", ...,
+// "00000000-0000-0000-0000-00000000000N"
```

Comment:
Great idea ! Having unique UUIDs instead of a generic  `00000000-0000-0000-0000-000000000000` is a good way to validate the results with some accuracy.

Reply:
Thank you.